### PR TITLE
Transfer shares

### PR DESCRIPTION
**Pull request by imyourm8**

### CODE REVIEW

**Comment by imyourm8** [comment]
I am not sure why tests are failing in CI, locally and in different PR they all passed.

File: contracts/staking/validatorShare/ValidatorShare.sol [hunk 1]
```diff
@@ -312,4 +330,17 @@ contract ValidatorShare is IValidatorShare, ERC20NonTransferable, OwnableLockabl
...
+        address from,
+        address to,
+        uint256 value
+    ) internal {
+        // get rewards for recipient 
+        _withdrawAndTransferReward(to);
+        // convert rewards to shares
+        (uint256 restakedShares, ) = _restake(from);
+        // move shares to recipient
+        super._transfer(from, to, value.add(restakedShares));
```
Comment (jdkanani) [inline contracts/staking/validatorShare/ValidatorShare.sol:344]: We chould just do 
```suggestion
        // get rewards for recipient 
        _withdrawAndTransferReward(to);
        // get rewards for sender 
        _withdrawAndTransferReward(from);
        // move shares to recipient
        super._transfer(from, to, value);
```

File: contracts/staking/validatorShare/ValidatorShare.sol [hunk 2]
```diff
@@ -312,4 +330,17 @@ contract ValidatorShare is IValidatorShare, ERC20NonTransferable, OwnableLockabl
...
+
+    function _transfer(
+        address from,
+        address to,
+        uint256 value
+    ) internal {
+        // get rewards for recipient 
+        _withdrawAndTransferReward(to);
+        // convert rewards to shares
+        (uint256 restakedShares, ) = _restake(from);
```
Comment (jdkanani) [inline contracts/staking/validatorShare/ValidatorShare.sol:342]:
```suggestion
        if (to == 0x0 || from == 0x0) {
            // get rewards for recipient 
            _withdrawAndTransferReward(to);
            // convert rewards to shares
            (uint256 restakedShares, ) = _restake(from);
        }
```

**Review by jdkanani — CHANGES_REQUESTED** [review]

File: contracts/staking/validatorShare/ValidatorShare.sol [hunk 3]
```diff
@@ -109,22 +109,30 @@ contract ValidatorShare is IValidatorShare, ERC20NonTransferable, OwnableLockabl
         return amountToDeposit;
     }
 
-    function restake() public {
-        uint256 liquidReward = _withdrawReward(msg.sender);
+    function restake() public returns(uint256, uint256) {
```
Comment (jdkanani) [inline contracts/staking/validatorShare/ValidatorShare.sol:112]: Do we need this change?

**Review by jdkanani — APPROVED** [review]

**Pull request merged**